doc/dev/scripts-and-sandbox.txt

```diff
@@ -0,0 +1,96 @@
+* There's a procedure for moving scripts from ``sandbox/`` into ``scripts/``.
+
+Read on!
+
+Sandbox script requirements and suggestions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+All scripts in ``sandbox/`` must:
+
+* be importable (enforced by ``test_import_all`` in
+  ``test_sandbox_scripts.py``)
+* be mentioned in ``sandbox/README.rst``
+* have a shebang line ``#! /usr/bin/env python``
+* be command-line executable (``chmod +x``)
+* have a Copyright message
+
+All *new* scripts being added to ``sandbox/`` should:
+
+* have decent automated tests
+* be used in a protocol (see khmer-protocols) or a recipe (see khmer-recipes)
+* be pep8 and pylint nice (see ``make pep8`` and ``make_diff_pylint``.
+
+Command line standard options for scripts/
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+All scripts in scripts/ should have the following options:
+
+* ``--force``
+* ``--loadtable`` and ``--savetable`` where appropriate (see khmer_args.py)
+
+Upgrading a script from 'sandbox' to 'scripts'
+----------------------------------------------
+
+First, everything needed (all library support code) should be already
+committed to khmer master after the usual review process; the relevant
+script(s) should be in ``sandbox/``.
+
+Second, an issue should be started explicitly to discuss whether the
+script(s) should be moved from ``sandbox/`` into ``scripts/``.  This issue
+should discuss the general need for this script, outside of a particular
+paper pipeline.  (Note that there is no imperative to move a script
+out of ``sandbox/``; if we think it's useful code to have around and
+want to keep it functioning, we should just add in automated tests and
+otherwise level it up.)
+
+Third, assuming we reach general agreement about moving the script(s)
+into ``scripts/``, start a pull request to do so, referencing the
+issue and containing the following checklist.  The PR should start by
+moving the script from ``sandbox/`` into ``scripts/``, and moving the
+tests out of the ``test_sandbox_scripts.py`` file.
+
+Last but not least, intensive code review may raise more general
+issues that could apply to the entire code base; if contentious or
+needing discussion, these issues may be punted to general issues so as
+to not block a merge.
+
+A checklist for moving a script into the scripts/ directory from sandbox/
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Copy or paste this checklist into the PR, in addition to the normal
+development/PR checklist::
+
+   - [ ] most or all lines of code are covered by automated tests (see output of ``make diff-cover``)
+   - [ ] ``make diff_pylint`` is clean
+   - [ ] the script has been updated with a ``get_parser()`` and added to doc/user/scripts.txt
+   - [ ] argparse help text exists, with an epilog docstring, with examples and options
+   - [ ] standard command line options are implemented
+   - [ ] version and citation information is output (`khmer_args.info`)
```

Comment:
version and citation information is output to STDERR ...
and all diagnostic info to STDERR